### ISSUE TITLE
There shouldn't be horizontal scroll space on mobile

### ISSUE THREAD

**Issue by david-christiansen** [leanprover/reference-manual#138]
There shouldn't be space to scroll to the right on mobile browsers, as in this screenshot:

![Screenshot_20241104-113213](https://github.com/user-attachments/assets/4f4b3791-6c9b-42d0-b662-1aacc1b70ccf)

**Comment by jakobvase** [comment]
Change `code, pre` to

```css
pre, code {
    font-family: var(--verso-code-font-family);
    font-variant-ligatures: none;
    overflow-x: scroll;
    padding-bottom: .5rem /* so the scroll bar doesn't cover the code */
}
```

and `.hl.lean.inline` to 

```css
.hl.lean.inline {
  display: inline;
  white-space: pre-line;
}
```

which makes it wrap the code, but otherwise keep it as is.

Finally we need to move the little link button into the box. Or remove it on mobile. But if we want to move it in, I would add the following to the inline css:

```css
/* below .namedocs .label, to move the label further left */
@media screen and (max-width: 700px) {
  .namedocs .label {
    right: 1.5rem;
  }
}
```

and the following to `book.css`

```css
/* after .permalink-widget.block */
@media screen and (max-width: 700px) {
    .permalink-widget.block {
        right: 0;
        opacity: 1;
    }
}
```

**Comment by david-christiansen** [comment]
I think this is _mostly_ fixed, but not 100%, by #148 - leaving this open until I've done a bit more investigation.

Thanks for the help!

**Comment by jakobvase** [comment]
It is. One issue that remains is that the headings are too big for small screens. The quick fix is adding an `overflow-x: hidden` to `h1, h2, h3, h4, h5, h6`. But I would rather pick a smaller font size for at least the `h1` and `h2` elements.

Another option is to add `word-break: break-word` to all headings, which causes them to break in the middle of words if necessary. I would add that as a fallback if some heading just has a really long word.

**Comment by jakobvase** [comment]
<img width="375" alt="Screenshot 2024-11-17 at 11 43 41" src="https://github.com/user-attachments/assets/33e051e6-c4d0-4c02-952f-1c104f83816b">

With break-word

**Comment by jakobvase** [comment]
There is also `hyphens: auto`, which adds hyphenation. It requires a `lang="en"` on `<html>` and is more fiddly, so I would go with break-word for now.

**Comment by david-christiansen** [comment]
Thanks!

I agree that the headers should shrink on small screens, and maybe even big ones, esp. now that there are explicit section numbers to help disambiguate visually-similar headers. Could you suggest some header font styles that go all the way down to `h6`, but with a smaller `h1`?

**Comment by jakobvase** [comment]
I would go with:

```css
h1 {
  font-size: 2.3rem;
  margin-bottom: 1.3rem;
}
h2 {
  font-size: 1.7rem;
}
h3 {
  font-size: 1.4rem; /* same as desktop, would also keep h4, h5 and h6 as desktop */
}
```

I don't think it provides value to differentiate between 6 levels of headings. In the language reference today, you use 4, and the 5th one is the same size as the body text, but bold, if you ever need it. If you want a sixth heading, I would go with `font-size: 1rem; font-weight: 500;`.

I also noticed that there is horizontal scrolling on the index page. That's solved by adding `overflow-x: scroll` to `main .theIndex .division > ol > li`.

One more thing about the index, I would differentiate between vertical and horizontal gap on the `ol` elements (`main .theIndex .division > ol`): `gap: 2rem 3rem`. On mobile I would actually just make it a flat list with `flex-direction: column; flex-wrap: nowrap; gap: 1rem;`.

**Comment by david-christiansen** [comment]
It's also possible to render the manual as one giant HTML file, which I use for proofreading, and there it has 6 levels of headings. Plus, this is all intended to be a flexible tool that others can write their documents with. So it is relevant to have something non-terrible for all of them, at least.

Thanks for the updates! Will test and get back to you.

**Comment by david-christiansen** [comment]
OK, I've inserted the mobile stylesheet fixes that you suggested.

Instead of the word wrap in the index, I had the code that makes the HTML insert soft hyphenation opportunities according to camelCase and namespace separators, which I think gives a better experience while reading here than the character-based wrapping would. I've verified that all the horizontal overflows were due to long identifiers and are now gone.
